### ISSUE TITLE
Syntax extension: inline function declaration with two or more parameters

### ISSUE THREAD

**Issue by fingolfin** [gap-system/gap#37]
For a long time, GAP allowed creating inline functions with a single parameter, e.g.
`f := x -> x + 1;`
Recently, as a feature backport from HPC-GAP, a similar feature for parameterless inline functions was added:
`g := -> 42;`
It woulde be nice if there was a similar syntax for functions with more than one parameter, e.g.
`h := (x,y) -> x+y;`
This can be used in many places, e.g. when sorting a list by a custom function:
`Sort(list, (x,y) -> 2*x < y);`
This examples also shows why I did not suggest this simpler syntax (as it woudl lead to ambiguities):
`h := x,y -> x+y;`
For consistency, I'd then also allow adding parentheses for the 1 and 0 parameter versions, i.e.
`f := (x) -> x + 1;`
and
`g := () -> 42;`

**Comment by stevelinton** [comment]
This might be quite hard to parse because it looks like the permutation syntax. Suppose that a1,a2, etc. were all defined global variables with distinct positive integer values then

x := (a1,a2,a3......a99,...a999,....

could be the start of an assignment of a permutation literal to x, or of an assignment of a function with
many arguments to x.  The parser is not really capable of unbounded lookahead of this kind.

If you can come up with a syntax that avoids this, then I would be happy to support it

**Comment by stevelinton** [comment]
Following discussion, here are some things that could work. Not clear any of them is nice enough to be worth the trouble:

```
`(x,y)-> <expr>
{x,y}-> <expr>
{x,y -> <expr> }
function x,y -> <expr>
& x,y -> <expr>
```

**Comment by fingolfin** [comment]
So in principle I am fine with any of these (though I'd prefer the shorter ones without 'function'). When using `{ }` my only concern is that we might want to use curly braces for other syntax extensions in the future, say for a hyptothetical built-tin dictionary / hash table type, as in `var := {x := 1, y := }`.  But based on discussion with Chris, this should still be possible, as we can relatively easily distinguish between such a dict and `{x,y}-> <expr>` or `{x,y -> <expr> }`.

Oh, and I'd recommend using the first one with a backtick, as the backtick now stands for immutable. The notation with ampersand is also fine by me (and it feels vaguely familiar... I wonder if I've seen that before), except that here, too, we should be careful before using up one of the few remaining "free" characters that are easily accesible from regular (western) keyboards.

**Comment by fingolfin** [comment]
@ChrisJefferson this is one of the issues we discussed earlier today

**Comment by alex-konovalov** [comment]
For the reference, this is now attempted by @ChrisJefferson in #490.

**Comment by fingolfin** [comment]
Resolved by PR #490
